diff --git a/node_modules/node-ical/node-ical.js b/node_modules/node-ical/node-ical.js
index 4503e19..b0bb291 100644
--- a/node_modules/node-ical/node-ical.js
+++ b/node_modules/node-ical/node-ical.js
@@ -1,5 +1,8 @@
-const fs = require('fs');
-const axios = require('axios');
+// Patched by Tricordarr to support React-Native
+// const fs = require('fs');
+const fs = require('react-native-fs');
+// const axios = require('axios');
+import axios from "axios";
 const ical = require('./ical.js');
 
 /**
