diff --git a/node_modules/react-native-paper/src/components/Menu/Menu.tsx b/node_modules/react-native-paper/src/components/Menu/Menu.tsx
index 55922c1..0921f18 100644
--- a/node_modules/react-native-paper/src/components/Menu/Menu.tsx
+++ b/node_modules/react-native-paper/src/components/Menu/Menu.tsx
@@ -89,6 +89,10 @@ export type Props = {
    * Inner ScrollView prop
    */
   keyboardShouldPersistTaps?: ScrollViewProps['keyboardShouldPersistTaps'];
+  /**
+   * Fires at most once per frame during scrolling of the inner ScrollView.
+   */
+  onScroll?: ScrollViewProps['onScroll'];
   /**
    * testID to be used on tests.
    */
@@ -194,6 +198,7 @@ const Menu = ({
   children,
   theme: themeOverrides,
   keyboardShouldPersistTaps,
+  onScroll,
 }: Props) => {
   const theme = useInternalTheme(themeOverrides);
   const insets = useSafeAreaInsets();
@@ -680,6 +685,7 @@ const Menu = ({
                 {(scrollableMenuHeight && (
                   <ScrollView
                     keyboardShouldPersistTaps={keyboardShouldPersistTaps}
+                    onScroll={onScroll}
                   >
                     {children}
                   </ScrollView>
